src/components/structures/MessagePanel.js

```diff
@@ -935,3 +963,62 @@ export default class MessagePanel extends React.Component {
         );
     }
 }
+
+/**
+ * Check for undecryptable messages that were sent while the user was not in
+ * the room.
+ *
+ * @param {Array<MatrixEvent>} events The timeline events to check
+ * @param {Room} room The room that the events are in
+ * @param {string} userId The user's ID
+ *
+ * @return {Number} The index within `events` of the event after the most recent
+ * undecryptable event that was sent while the user was not in the room.  If no
+ * such events were found, then it returns 0.
+ */
+function checkForPreJoinUISI(events, room, userId) {
+    if (events.length === 0 ||
+        !MatrixClientPeg.get().isRoomEncrypted(room.roomId)) {
+        return 0;
+    }
+
+    let i;
+    let userMembership = "leave";
+    // find the last event that is in a timeline (events may not be in a
+    // timeline if they have not been sent yet) and get the user's membership
+    // at that point.
+    for (i = events.length - 1; i >= 0; i--) {
+        const timeline = room.getTimelineForEvent(events[i].getId());
+        if (timeline) {
+            const userMembershipEvent =
+                  timeline.getState(EventTimeline.FORWARDS).getMember(userId);
+            userMembership = userMembershipEvent ? userMembershipEvent.membership : "leave";
+            const timelineEvents = timeline.getEvents();
+            for (let j = timelineEvents.length - 1; j >= 0; j--) {
```

Comment:
Why is this a loop? We don't seem to use `j` inside...?

Reply:
Because I'm silly and used the wrong index.